x/packages/keeper/package_versions_storage.go

```diff
@@ -0,0 +1,349 @@
+		return nil, utils.LavaError(ctx, k.Logger(ctx), "get_package_from_package_entry", map[string]string{"err": err.Error()}, "could not get package from package fixation entry")
+	}
+
+	return unmarshaledPackage, nil
+}
+
+// Function to get a package's latest version
+func (k Keeper) GetPackageLatestVersion(ctx sdk.Context, packageIndex string) (*types.Package, error) {
+	// get the requested package's latest fixated entry
+	latestVersionPackageFixationEntry, found := common.GetEntryFromStorage(ctx, packageIndex, math.MaxUint64, k.GetFixationEntryByIndex)
+	if !found {
+		return nil, utils.LavaError(ctx, k.Logger(ctx), "get_package_by_epoch", map[string]string{"packageIndex": packageIndex}, "could not get packageFixationEntry with index and epoch")
+	}
+
+	// get the package from the package fixation entry
+	latestVersionPackage, err := k.getPackageFromPackageFixationEntry(ctx, latestVersionPackageFixationEntry)
+	if err != nil {
+		return nil, utils.LavaError(ctx, k.Logger(ctx), "get_package_from_package_entry", map[string]string{"err": err.Error()}, "could not get package from package fixation entry")
+	}
+
+	return latestVersionPackage, nil
+}
+
+// Function to extract and unmarshal the package object from a package fixation entry
+func (k Keeper) getPackageFromPackageFixationEntry(ctx sdk.Context, packageFixationEntry *commontypes.Entry) (*types.Package, error) {
+	// get the marshaled data from the package fixation entry
+	marshaledPackage := packageFixationEntry.GetMarshaledData()
+
+	// unmarshal the marshaled data to get the package object
+	var unmarshaledPackage types.Package
+	err := k.cdc.Unmarshal(marshaledPackage, &unmarshaledPackage)
+	if err != nil {
+		return nil, utils.LavaError(ctx, k.Logger(ctx), "unmarshal_package", map[string]string{"err": err.Error()}, "could not unmarshal package")
+	}
+
+	return &unmarshaledPackage, nil
+}
+
+// Function to add a new package to the packageVersionsStorage. It supports addition of packages with new index and packages with existing index (index that is already saved in the storage)
+func (k Keeper) AddNewPackageToStorage(ctx sdk.Context, packageToAdd *types.Package) error {
+	// get current epoch and next epoch
+	currentEpoch := k.epochStorageKeeper.GetEpochStart(ctx)
+
+	// make the package's epoch field's value be the next epoch (which is when it will be available for purchase)
+	nextEpoch, err := k.epochStorageKeeper.GetNextEpoch(ctx, currentEpoch)
+	if err != nil {
+		return utils.LavaError(ctx, k.Logger(ctx), "get_next_epoch", map[string]string{"err": err.Error()}, "could not get next epoch")
+	}
+	packageToAdd.Epoch = nextEpoch
+
+	// get current epochBlocks
+	epochBlocks, err := k.epochStorageKeeper.EpochBlocks(ctx, currentEpoch)
+	if err != nil {
+		return utils.LavaError(ctx, k.Logger(ctx), "epoch_blocks", map[string]string{"err": err.Error()}, "could not get epoch blocks")
+	}
+
+	// check that the duration is at least an epoch
+	if packageToAdd.GetDuration() < epochBlocks {
```

Comment:
remove this checks, duration is going to be a value bigger than 0 in months
also ComputeUnitsPerEpoch  will need to be calculated and be set in the proposal, not calculated (there was a design change)

Reply:
done